Adding the application Flower

adityapgupta opened the pull request:
Current progress for adding Flower to AIOpsLab using a k8s cluster

adityapgupta commented:
@microsoft-github-policy-service agree company="Indian Institute of Science"

JacksonArthurClark commented:
Converted to a draft since this is still a WIP. When ready for review just mark as 'Ready for Review' and add me as a reviewer :)

HacksonClark commented:
@adityapgupta Is this still a draft?

adityapgupta commented:
This is almost complete, I just need to add the readmes and see if I can work around the sleep issue. I have fast forwarded the aiopslab-applications change too

adityapgupta commented:
Hey @HacksonClark , I have removed the use of sleeps in the fault injection (sorry for the delay). Need to make a few small changes now, would appreciate ur advice here:

1. Flower doesnt use stuff like Prometheus and OpenEBS. Turning all of them off everywhere for Flower is a bit inconvenient. Is it ok if we expect the users to have the kind cluster running for this?
2. I have added a LLaMa client for my testing which uses a Groq client. Do you want it to stay this way or should I swap it out for something else?
3. Where should I update the relevant requirements?
4. Which readmes should I update for deployment and testing details? I have already added a small one about the Flower application itself.

adityapgupta commented:
@HacksonClark 

1. I dont mean kind specifically, I dont even use k8s here. Its just that I need to write an if condition everywhere to not launch/search for the k8s related stuff while using a docker application. But considering the main readme asks users to setup kind anyway, I thought I can skip this check for docker applications
2. Ok Ill keep the Groq client in that case for diversity (and for people like me who don't have a paid OpenAI key)
3. Ill update pyproject then
4. Will make a mention in the main readme about docker and Flower

adityapgupta commented:
@gaganso I have resolved the merge conflicts and fast forwarded the branch. @HacksonClark my commit of aiopslab-applications seems to be behind the current one, but the current one doesn't have the `flower` folder. Let me know how to fix this

HacksonClark commented:
@adityapgupta I wonder if I accidentally overwrote it when working on `aiopslab-applications` (we're using it for a new project). Just add it as a submodule and submit a PR.

adityapgupta commented:
@HacksonClark `aiopslab-applications` has been updated and fast-forwarded accordingly. Feel free to merge now
